satpy/tests/reader_tests/test_ahi_gridded.py

```diff
@@ -0,0 +1,262 @@
+            np.testing.assert_allclose(res, da.from_array(test_arr))
+
+    @mock.patch('satpy.readers.ahi_gridded.AHIGriddedFileHandler._read_data')
+    def test_get_dataset(self, mocked_read):
+        """Check that a good dataset is returned on request."""
+        m = mock.mock_open()
+
+        out_data = np.array([[100., 300., 500.],
+                             [800., 1500., 2040.]])
+        mocked_read.return_value = out_data
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            res = self.fh.get_dataset(self.key, self.info)
+            mocked_read.assert_called()
+            # Check output data is correct
+            np.testing.assert_allclose(res.values, out_data)
+            # Also check a couple of attributes
+            self.assertEqual(res.attrs['name'], self.key['name'])
+            self.assertEqual(res.attrs['wavelength'], self.info['wavelength'])
+
+
+class TestAHIGriddedLUTs(unittest.TestCase):
+    """Test case for the downloading and preparing LUTs."""
+    def setUp(self):
+        """Create a test file handler."""
+        m = mock.mock_open()
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            in_fname = 'test_file'
+            fh = AHIGriddedFileHandler(in_fname,
+                                       {'area': 'fld'},
+                                       filetype_info={'file_type': 'tir.01'})
+            self.fh = fh
+
+        key = {'calibration': 'counts',
+               'name': 'vis.01'}
+        info = {'units': 'unitless',
+                'standard_name': 'vis.01',
+                'wavelength': 10.8,
+                'resolution': 0.05}
+        self.key = key
+        self.info = info
+
+    @mock.patch('ftplib.FTP')
+    def test_download_luts(self, mock_ftp):
+        """Test that the FTP library is called for downloading LUTS."""
+        m = mock.mock_open()
+        with mock.patch('satpy.readers.ahi_gridded.open', m, create=True):
+            self.fh._download_luts('/test_file')
+            mock_ftp.assert_called()
+
+    @mock.patch('os.remove')
+    def test_untar_luts(self, mock_remove):
+        """Test that the untar library is used correctly."""
+        m = mock.MagicMock()
+        with mock.patch('tarfile.open', m, create=True):
+            self.fh._untar_luts('/test_file', '/test_dir/')
+            m.assert_called()
+            mock_remove.assert_called()
+
+    @mock.patch('os.path.join')
+    @mock.patch('shutil.rmtree')
+    @mock.patch('shutil.move')
+    @mock.patch('satpy.readers.ahi_gridded.AHIGriddedFileHandler._untar_luts')
+    @mock.patch('satpy.readers.ahi_gridded.AHIGriddedFileHandler._download_luts')
+    @mock.patch('tempfile.gettempdir')
+    @mock.patch('pathlib.Path')
+    def test_get_luts(self, *mocks):
+        """Check that the function to download LUTs operates successfully."""
+        self.fh._get_luts()
+        mocks[2].assert_called()
+        mocks[3].assert_called()
```

Comment:
This test just checks that `_untar_luts` and `_download_luts` are called. To check that the lut download operates successfully, shouldn't we be checking that a downloaded tar file is correctly untarred and place in the right directory?